src/crimson/osd/main_config_bootstrap_helpers.cc

```diff
@@ -84,6 +84,54 @@ seastar::future<> populate_config_from_mon()
   });
 }
 
+struct first_double {
+  double value;
+  first_double() : value(0.0) {}
+  first_double(double v) : value(v) {}
+};
+struct second_double {
+  double value;
+  second_double() : value(0.0) {}
+  second_double(double v) : value(v) {}
+};
```

Comment:
Does this mean that with each new option we add we will need to add a new struct e.g `third_double`?
I would strongly prefer not introducing a new struct for each option given that we would want the ability to support lots of options eventually.
Moreover, not introducing a new type for each option would help as to prevent `std::decay_t<decltype(value)>` usage to identify which option is being set.

It seems like we use `std::to_string` to emplace the options into `early_args`. Why should we support each new type if in the end they would all be converted to strings?

Reply:
I will think more about it if we can make everyone string.